experimental/compactdecimal/src/provider.rs

```diff
@@ -0,0 +1,104 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+// Provider structs must be stable.
+#![allow(clippy::exhaustive_structs, clippy::exhaustive_enums)]
+// Suppress a warning on zerovec::makevarule.
+#![allow(missing_docs)]
+
+//! Data provider struct definitions for this ICU4X component.
+//!
+//! Read more about data providers: [`icu_provider`]
+
+use alloc::borrow::Cow;
+use icu_provider::{yoke, zerofrom};
+use zerovec::ZeroMap2d;
+
+/// Relative time format V1 data struct.
+
+#[icu_provider::data_struct(
+    LongCompactDecimalFormatDataV1Marker = "compactdecimal/long@1",
+    ShortCompactDecimalFormatDataV1Marker = "compactdecimal/short@1"
+)]
+#[derive(Debug, Clone, Default, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[cfg_attr(
+    feature = "datagen", 
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_compactdecimal::provider)
+)]
+#[yoke(prove_covariance_manually)]
+pub struct CompactDecimalPatternDataV1<'data> {
+    /// A map keyed on log10 of the CLDR `type` attribute and the `cldr` count attribute.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub patterns: ZeroMap2d<'data, i8, Count, PatternULE>,
+}
+
+/// A CLDR plural keyword, or the explicit value 1.
+/// See <https://www.unicode.org/reports/tr35/tr35-numbers.html#Language_Plural_Rules>.
+#[zerovec::make_ule(CountULE)]
+#[zerovec::derive(Debug)]
+#[derive(Copy, Clone, PartialOrd, Ord, PartialEq, Eq, Debug)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[cfg_attr(
+    feature = "datagen", 
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_compactdecimal::provider)
+)]
+#[repr(u8)]
+pub enum Count {
+    /// The CLDR keyword `other`.
+    Zero = 0,
+    /// The CLDR keyword `one`.
+    One = 1,
+    /// The CLDR keyword `two`.
+    Two = 2,
+    /// The CLDR keyword `few`.
+    Few = 3,
+    /// The CLDR keyword `many`.
+    Many = 4,
+    /// The CLDR keyword `other`.
+    Other = 5,
+    /// The explicit 1 case, see <https://www.unicode.org/reports/tr35/tr35-numbers.html#Explicit_0_1_rules>.
+    Explicit1 = 6,
+    // NOTE(egg): No explicit 0, because the compact decimal pattern selection
```

Comment:
Event there is no `0`, I prefer to call it `Explicit01` or something like that because this case for 0 and 1. because the algorithm can change anytime or any exception can be added.

Reply:
Explicit 0 would be a different alternative. You could have ExplicitValue(i32) or somesuch, then that is no longer a byte-sized enum. Anyway, as Shane wrote, let’s see where icu_pattern goes.

Reply:
okay, let's see where icu_pattern goes.